### PR TITLE
docs(eslint-plugin): add examples to signal-store-feature-should-use-generic-type.md

### LINKED ISSUE #4502: Add documentation for the ESLint rule signal-store-feature-should-use-generic-type
### Information

The rule `signal-store-feature-should-use-generic-type` should be documented at https://github.com/ngrx/platform/blob/main/projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md.

It should also contain one or more examples showcasing some "bad" and "good" snippets.
For some examples we can take a look at the test cases at https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/spec/rules/signals/signal-store-feature-should-use-generic-type.spec.ts

### Documentation page

https://ngrx.io/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by paweltwardziak**
+ align prefer-protected-state.md
Closes #4502

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

See #4502

## What is the new behavior?

See #4502

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Closes #4502

### CODE REVIEW

**Comment by paweltwardziak** [comment]
@timdeschryver ready for CRing :)

File: projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md [hunk 1]
```diff
@@ -10,3 +10,177 @@ A custom Signal Store feature that accepts an input should define a generic type
...
+  );
+  return feature;
+}
+```
+
+Examples of **correct** code for this rule:
+
+<!-- prettier-ignore -->
+```ts
+// GOOD ✅
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md:116]: I think we can remove the `Good` and `Problem` lines in the examples.
There's already a header that indicates the good and bad examples.

For the bad examples, it's important to indicate what the error is, which is what we do with `~~~~~` - this is the most important.

What do you think?
If you agree, we can also remove remove it for prefer-protected-state.

**Review by timdeschryver — COMMENTED** [review]
Thanks for the PR @paweltwardziak, I left some comments.
Let me know what you think

File: projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md (hunk 1 above)
Comment (paweltwardziak) [reply projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md:116]: Yes, this makes sense. I made kinda duplication. Will remove Good/Problem 👍

File: projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md [hunk 2]
```diff
@@ -10,3 +10,177 @@ A custom Signal Store feature that accepts an input should define a generic type
 
 <!-- Everything above this generated, do not edit -->
 <!-- MANUAL-DOC:START -->
+
+## Rule Details
+
+This rule ensure that Signal Store feature that accepts an input defines a generic type.
```
[the remark this answers is not in the document]
Comment (paweltwardziak) [reply projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md:16]: Will do, good point 👍

File: projects/ngrx.io/content/guide/signals/signal-store/index.md [hunk 3]
```diff
@@ -222,6 +222,8 @@ For more details on the `patchState` function, refer to the [Updating State](/gu
 
 </div>
 
+### SignalStore's state protection
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/signals/signal-store/index.md:225]: Can you revert this change please?

File: projects/ngrx.io/content/guide/eslint-plugin/rules/prefer-protected-state.md [hunk 4]
```diff
@@ -17,28 +17,32 @@ This rule ensures that state changes are only managed by the Signal Store to pre
...
-// GOOD ✅
 const Store = signalStore(
   { protectedState: true },
   withState({}),
 );
 ```
+
+## Further reading
+
+- [SignalStore's state protection](guide/signals/signal-store#signalstores-state-protection)
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/eslint-plugin/rules/prefer-protected-state.md:48]: Keep in mind that when we remove the "### SignalStore's state protection" header, that we also need to update this link (or remove it).
